src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -0,0 +1,361 @@
+     * @param relevantArtifactModesWithoutComplements ArtifactModes to filter.  Do not include complements. Never {@code null}.
+     * @param preAdapterQAnnotatedVariants Variant contexts that have already been annotated by {@link OrientationBiasFilterer#annotateVariantContextWithPreprocessingValues(VariantContext, SortedSet, Map)} Never {@code null}.
+     * @param preAdapterQScoreMap Mapping from ArtifactMode to the preAdapterQ score.  relevantArtifactModes should be included as keys, but not required. Never {@code null}.
+     * @return The same variant contexts with the genotypes filter field populated with orientation bias filtering results..
+     */
+    public static List<VariantContext> annotateVariantContextsWithFilterResults(final double fdrThresh, final SortedSet<ArtifactMode> relevantArtifactModesWithoutComplements, final List<VariantContext> preAdapterQAnnotatedVariants,
+                                                                                final Map<ArtifactMode, Double> preAdapterQScoreMap) {
+
+        // This holds the relevantArtifact modes and the complements, since we need to filter both.
+        final SortedSet<ArtifactMode> relevantArtifactModes = new TreeSet<>();
+        relevantArtifactModes.addAll(relevantArtifactModesWithoutComplements);
+        relevantArtifactModes.addAll(OrientationBiasUtils.createReverseComplementArtifactModes(relevantArtifactModesWithoutComplements));
+
+        final List<String> sampleNames = preAdapterQAnnotatedVariants.get(0).getSampleNamesOrderedByName();
+        final Map<String, SortedMap<Genotype, VariantContext>> sampleNameToVariants = createSampleToGenotypeVariantContextSortedMap(sampleNames, preAdapterQAnnotatedVariants);
+
+        // This map will hold all updated genotypes (across samples)
+        final Map<VariantContext, List<Genotype>> newGenotypes = new HashMap<>();
+
+        // For each sample, perform the actual filtering.
+        for (final String sampleName: sampleNames) {
+
+            // Count the total number of unfiltered genotypes for this sample, regardless of artifact mode or not.
+            //  I.e. only remove filtered or ref/ref genotypes and count the rest.
+            final long unfilteredGenotypeCount = OrientationBiasUtils.calculateUnfilteredNonRefGenotypeCount(preAdapterQAnnotatedVariants, sampleName);
+
+            final SortedMap<Genotype, VariantContext> genotypesToConsiderForFiltering = sampleNameToVariants.get(sampleName);
+
+            // Save some time, especially for the normal sample
+            if (genotypesToConsiderForFiltering.keySet().size() == 0) {
+                logger.info(sampleName + ": Nothing to filter.");
+                continue;
+            }
+
+            // Populate counts of artifact modes.  These are genotypes that we can potentially cut.
+            final Map<ArtifactMode, Long> artifactModeCount = createArtifactModeCountMap(relevantArtifactModes, genotypesToConsiderForFiltering);
```

Comment:
Did not make it an EnumMap, but removed ArtifactMode entirely and using Transition.

Done